gammapy/background/tests/test_models.py

```diff
@@ -132,6 +137,106 @@ def make_test_array(empty=True):
     return multi_array
 
 
+def make_test_array_fillobs(excluded_sources=None, fov_radius=Angle(2.5, "deg")):
+    dir = str(gammapy_extra.dir) + '/datasets/hess-crab4-hd-hap-prod2'
+    data_store = DataStore.from_dir(dir)
+    obs_table = data_store.obs_table
+    multi_array = make_test_array()
+    multi_array.fill_obs(obs_table, data_store)
+    return multi_array
+
+
+def make_test_array_oneobs(excluded_sources=None, fov_radius=Angle(2.5, "deg")):
+    dir = str(gammapy_extra.dir) + '/datasets/hess-crab4-hd-hap-prod2'
+    data_store = DataStore.from_dir(dir)
+    obs_table = data_store.obs_table
+    multi_array = make_test_array()
+    multi_array.fill_obs([obs_table[0]], data_store, excluded_sources, fov_radius)
+    return multi_array, data_store, obs_table[0]
+
+
+def make_excluded_sources():
+    centers = SkyCoord([1, 0], [2, 1], unit='deg')
+    radius = Angle('0.3 deg')
+    sources = SkyCircleRegion(pos=centers, radius=radius)
+    catalog = Table()
+    catalog["RA"] = sources.pos.data.lon
+    catalog["DEC"] = sources.pos.data.lat
+    catalog["Radius"] = sources.radius
+    return catalog
+
+
+def make_source_nextCrab():
+    center = SkyCoord([84, 89], [23, 20], unit='deg', frame='icrs')
+    radius = Angle('0.3 deg')
+    sources = SkyCircleRegion(pos=center, radius=radius)
+    catalog = Table()
+    catalog["RA"] = sources.pos.data.lon
+    catalog["DEC"] = sources.pos.data.lat
+    catalog["Radius"] = sources.radius
+    return catalog
+
+
+def test_compute_pie_fraction():
+    excluded_sources = make_excluded_sources()
+    pointing_position = SkyCoord(0.5, 0.5, unit='deg')
+    # Test that if the sources are out of the fov, it gives a pie_fraction equal to zero
+    pie_fraction = compute_pie_fraction(excluded_sources, pointing_position, Angle(0.3, "deg"))
+    assert_allclose(pie_fraction, 0)
+
+    # I have to use an other object excluded_region because the previous one was
+    # already sorted in the compute_pie_fraction
+    excluded_sources2 = make_excluded_sources()
+    source_closest = SkyCoord(excluded_sources2["RA"][1], excluded_sources2["DEC"][1], unit="deg")
+    separation = pointing_position.separation(source_closest).value
+    pie_fraction = compute_pie_fraction(excluded_sources, pointing_position, Angle(5, "deg"))
+    pie_fraction_expected = (2 * np.arctan(excluded_sources2["Radius"][1] / separation) / (2 * np.pi))
+    assert_allclose(pie_fraction, pie_fraction_expected)
+
+
+def test_select_events_outside_pie():
```

Comment:
And if you're not reading the image from the Gammapy tests, I also don't care much about the image format.
TIFF, PNG, ... is fine.

I just don't want this as part of the official tests, because a TIFF image reader isn't available for every user that wants to run the tests.